src/filepattern/cpp/interface/filepattern.h

```diff
@@ -251,13 +252,13 @@ class FilePatternFactory {
 
         PatternObject* getObject(const std::string& path, const std::string& file_pattern, const std::string& block_size, bool recursive, bool suppressWarnings) {
             if (block_size == "") {
-                if(s::endsWith(path, ".txt")) {
+                if(std::filesystem::is_regular_file(path)) {
```

Comment:
See the example in Nyxus about how we bring in `std::filesystem`. Some compilers don't have it yet and have the experimental one.